runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -347,6 +337,76 @@ public void onPlayerMenuOptionClicked(PlayerMenuOptionClicked event)
 		}
 	}
 
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged gameStateChanged)
+	{
+		if (gameStateChanged.getGameState() == GameState.LOGGED_IN)
+		{
+			if (lastWorld != client.getWorld())
+			{
+				updateList();
+				lastWorld = client.getWorld();
+			}
+		}
+	}
+
+	@Subscribe
+	public void onWorldListLoad(WorldListLoad worldListLoad)
+	{
+		HashMap<Integer, Integer> worldData = new HashMap<>();
+
+		for (net.runelite.api.World w : worldListLoad.getWorlds())
+		{
+			worldData.put(w.getId(), w.getPlayerCount());
+		}
+
+		panel.updateListData(worldData);
+	}
+
+	/**
+	 * This method resets the scheduled timer, this will also fire the fetch worlds method that updates the list with
+	 * updated data.
+	 */
+	public void refresh()
+	{
+		if (worldResultFuture != null)
+		{
+			worldResultFuture.cancel(true);
+		}
+
+		worldResultFuture = executorService.scheduleAtFixedRate(() ->
```

Comment:
Just schedule this at startup. This is single-threaded executor, just do executorService.submit(this::fetchWorlds) here, no reason to restart entire task

Reply:
The goal is to reset the timer everything we want to manually refresh, would that achieve that goal?